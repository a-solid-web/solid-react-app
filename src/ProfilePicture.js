import React from "react";
import Image from "react-bootstrap/Image"


export const ProfilePicture = (props) => {
        return (
            <Image src={props.picture} fluid/>
        )
    }

