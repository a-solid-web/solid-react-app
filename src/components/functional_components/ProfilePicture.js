import React from "react";
import Image from "react-bootstrap/Image";

const ProfilePicture = (props) => {
  return (
    <Image
      style={{
        border: "solid #999 5px",
        borderRadius: 5,
        marginRight: 10,
        marginBottom: 10
      }}
      src={props.picture}
      fluid
    />
  );
};

export default ProfilePicture;