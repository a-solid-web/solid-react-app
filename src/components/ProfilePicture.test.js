import React from "react";
import ProfilePicture from "./ProfilePicture";
import renderer from "react-test-renderer";

describe('ProfilePicture matches snapshot when', () => {
    test('no props are passed.', () => {
        const component = renderer.create(
          <ProfilePicture />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    test('props are passed', () => {
        const component = renderer.create(
          <ProfilePicture picture="/hereisanad"/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})