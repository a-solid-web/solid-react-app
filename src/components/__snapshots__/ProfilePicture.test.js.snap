// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePicture matches snapshot when no props are passed. 1`] = `
<img
  className="img-fluid"
  style={
    Object {
      "border": "solid #999 5px",
      "borderRadius": 5,
      "marginBottom": 10,
      "marginRight": 10,
    }
  }
/>
`;

exports[`ProfilePicture matches snapshot when props are passed 1`] = `
<img
  className="img-fluid"
  src="/hereisanad"
  style={
    Object {
      "border": "solid #999 5px",
      "borderRadius": 5,
      "marginBottom": 10,
      "marginRight": 10,
    }
  }
/>
`;
